devito/operator/operator.py

```diff
@@ -663,8 +647,11 @@ def _soname(self):
 
     @cached_property
     def ccode(self):
-        from devito.ir.iet.visitors import CGen
-        return CGen(compiler=self._compiler).visit(self)
+        try:
+            return self._ccode_handler(compiler=self._compiler).visit(self)
+        except (AttributeError, TypeError):
+            from devito.ir.iet.visitors import CGen
+            return CGen(compiler=self._compiler).visit(self)
```

Comment:
Note for me from other PR, cannot be made class default due to circular imports.